extensions/map-layers-auth/src/ArcGis/ArcGisAccessClient.ts

```diff
@@ -274,7 +274,7 @@ export class ArcGisAccessClient implements MapLayerAccessClient {
 
     const endpointStr = (endpoint === ArcGisOAuth2EndpointType.Authorize ? "authorize" : "token");
     const urlObj = new URL(url);
-    if (urlObj.hostname.toLowerCase().endsWith("arcgis.com")) {
+    if (urlObj.hostname.toLowerCase() === "arcgis.com") {
```

Comment:
this doesnt look right

@mdastous-bentley

Reply:
This is more a way to speed up the endpoint url discovery, en though the arcgis.com url changes, the code should still work,  Do you want to get rid of it in this PR anyway?

Reply:
ok if it works then no problem, just wanted to make sure you reviewed the changes.

Reply:
oh yea, I dont think this change will work, I need to do more testing.

Reply:
What is the motivation of changing this?

Reply:
Hi, I created a comment last week (at the end) that codeQL was warning on the vulnerabilities. I reverted the change as master for now. I am new to iTwin codebase and am not sure to create a list of hostnames as codeQL suggested. Please let me know if there is a possible solution or the warning is not valid. Thanks.

https://github.com/iTwin/itwinjs-core/pull/3328#:~:text=I%20reverted%20this%20line%20as,sanitization%0AIncomplete%20URL%20substring%20sanitization

Reply:
Hi, I reverted this file as master. I made the change as per codeQL warning which suggests to create a list of allowed hosts. I am new to iTwin and am not sure with the list of host name to create. Please let me know if there is a possible solution or the warning is invalid. 

https://github.com/iTwin/itwinjs-core/pull/3328#discussion_r938076759